api/src/test/java/com/epam/pipeline/controller/user/UserControllerTest.java

```diff
@@ -0,0 +1,504 @@
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldExportUsers() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(userExportVO);
+        doReturn(TEST_ARRAY).when(mockUserApiService).exportUsers(userExportVO);
+
+        final MvcResult mvcResult = performRequest(post(USER_EXPORT_URL).content(content),
+                EXPECTED_CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM_VALUE);
+
+        verify(mockUserApiService).exportUsers(userExportVO);
+        assertFileResponse(mvcResult, "users.csv", TEST_ARRAY);
+    }
+
+    @Test
+    public void shouldFailExportUsers() {
+        performUnauthorizedRequest(post(USER_EXPORT_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadUsersByGroup() {
+        doReturn(pipelineUserList).when(mockUserApiService).loadUsersByGroup(TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(get(GROUP_URL).params(multiValueMapOf(GROUP, TEST_STRING)));
+
+        verify(mockUserApiService).loadUsersByGroup(TEST_STRING);
+        assertResponse(mvcResult, pipelineUserList, UserCreatorUtils.PIPELINE_USER_LIST_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadUsersByGroup() {
+        performUnauthorizedRequest(get(GROUP_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldFindGroups() {
+        doReturn(TEST_STRING_LIST).when(mockUserApiService).findGroups(TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(get(GROUP_FIND_URL).params(multiValueMapOf(PREFIX, TEST_STRING)));
+
+        verify(mockUserApiService).findGroups(TEST_STRING);
+        assertResponse(mvcResult, TEST_STRING_LIST, CommonCreatorConstants.STRING_LIST_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailFindGroups() {
+        performUnauthorizedRequest(get(GROUP_FIND_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUpsertGroupBlockingStatusByPostRequest() {
+        doReturn(groupStatus).when(mockUserApiService).upsertGroupBlockingStatus(TEST_STRING, true);
+
+        final MvcResult mvcResult = performRequest(post(String.format(GROUP_NAME_BLOCK_URL, TEST_STRING))
+                .params(multiValueMapOf(BLOCK_STATUS, true)));
+
+        verify(mockUserApiService).upsertGroupBlockingStatus(TEST_STRING, true);
+        assertResponse(mvcResult, groupStatus, UserCreatorUtils.GROUP_STATUS_INSTANCE_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUpsertGroupBlockingStatusByPutRequest() {
+        doReturn(groupStatus).when(mockUserApiService).upsertGroupBlockingStatus(TEST_STRING, true);
+
+        final MvcResult mvcResult = performRequest(put(String.format(GROUP_NAME_BLOCK_URL, TEST_STRING))
+                .params(multiValueMapOf(BLOCK_STATUS, true)));
+
+        verify(mockUserApiService).upsertGroupBlockingStatus(TEST_STRING, true);
+        assertResponse(mvcResult, groupStatus, UserCreatorUtils.GROUP_STATUS_INSTANCE_TYPE);
+    }
```

Comment:
Let's use a single methods to test these calls. See `ServerlessConfigurationControllerTest.shouldRun` for a similar case.